client/packages/invoices/src/Prescriptions/DetailView/Toolbar.tsx

```diff
@@ -49,11 +54,18 @@ export const Toolbar: FC = () => {
   });
 
   const handleDateChange = async (newPrescriptionDate: Date | null) => {
+    const currentDateValue = dateValue; // Revert to this value if user cancels
+
     if (!newPrescriptionDate) return;
+    setDateValue(newPrescriptionDate);
 
     const oldPrescriptionDate = DateUtils.getDateOrNull(prescriptionDate);
 
-    if (newPrescriptionDate === oldPrescriptionDate) return;
+    if (
+      newPrescriptionDate.toLocaleDateString() ===
+      oldPrescriptionDate?.toLocaleDateString()
+    )
```

Comment:
This was the erroneous date equality check. Changed to just a string comparison of the date value.